source-kafka/src/connector.rs

```diff
@@ -39,3 +42,69 @@ pub enum Command {
         state: Option<PathBuf>,
     },
 }
+
+impl Command {
+    pub fn execute<C: Connector>(&self) -> color_eyre::Result<()> {
+        let mut stdout = Box::new(LineWriter::new(stdout()));
```

Comment:
Ah, so it does. I can also make sure to lock stdout at the top level and skip the mutex throughout the rest of the program since we're explicitly passing stdout as a `Write` handle to the connector functions.